tests/test_runtime.rs

```diff
@@ -11,23 +13,74 @@ use primitives::serialize::Decode;
 use primitives::transaction::{
     AddKeyTransaction, AsyncCall, Callback, CallbackInfo, CallbackResult, CreateAccountTransaction,
     DeleteKeyTransaction, DeployContractTransaction, FunctionCallTransaction, ReceiptBody,
-    ReceiptTransaction, SwapKeyTransaction, TransactionBody, TransactionStatus,
+    ReceiptTransaction, SwapKeyTransaction, TransactionBody, TransactionStatus, FinalTransactionStatus
 };
 use primitives::types::AccountingInfo;
-use testlib::node::{Node, RuntimeNode, ShardClientNode};
+use std::sync::Arc;
+use std::sync::atomic::{AtomicU16, Ordering};
+use testlib::node::{
+    create_nodes_with_id_type, Node, NodeConfig, RuntimeNode,
+    ShardClientNode, ThreadNode, TEST_BLOCK_FETCH_LIMIT,
+};
+use testlib::test_helpers::{heavy_test, wait};
 use testlib::user::User;
 
 const POISONED_LOCK_ERR: &str = "The lock was poisoned.";
+const NUM_TEST_NODE: usize = 4;
+static TEST_PORT: AtomicU16 = AtomicU16::new(6000);
+
+fn test_chain_spec() -> ChainSpec {
+    ChainSpec::testing_spec(DefaultIdType::Named, 3, 3, AuthorityRotation::ProofOfAuthority).0
+}
 
 fn create_shard_client_node() -> ShardClientNode {
-    let client_cfg = ClientConfig::default_devnet();
+    let mut client_cfg = ClientConfig::default_devnet();
+    client_cfg.chain_spec = test_chain_spec();
     ShardClientNode::new(client_cfg)
 }
 
 fn create_runtime_node() -> RuntimeNode {
     RuntimeNode::new(&alice_account())
 }
 
+fn create_testnet_node(test_prefix: &str, test_port: u16) -> Vec<ThreadNode> {
+    let proxy_handlers: Vec<Arc<ProxyHandler>> = vec![Arc::new(BenchmarkHandler::new())];
+
+    let (_, account_names, mut nodes) = create_nodes_with_id_type(
+        NUM_TEST_NODE,
+        test_prefix,
+        test_port,
+        TEST_BLOCK_FETCH_LIMIT,
+        proxy_handlers,
+        DefaultIdType::Named,
+    );
+    assert_eq!(account_names[0], alice_account());
+    let mut nodes: Vec<_> = nodes
+        .drain(..)
```

Comment:
```suggestion
        .into_iter()
```
Use `drain` when you are planning to reuse the container, otherwise use `into_iter`.